components/scream/src/physics/p3/micro_p3_iso_c.f90

```diff
@@ -293,6 +293,15 @@ subroutine get_rain_dsd2_c(qr,nr,mu_r,lamr,cdistr,logn0r,rcldm) bind(C)
     call get_rain_dsd2(qr,nr,mu_r,lamr,cdistr,logn0r,rcldm)
   end subroutine get_rain_dsd2_c
 
+  subroutine rain_immersion_freezing_c(t,lamr,mu_r,cdistr,qr_incld,qrheti,nrheti) bind(C)
+
+      use micro_p3, only: rain_immersion_freezing
+      real(kind=c_real), value, intent(in) :: t, lamr, mu_r, cdistr, qr_incld
+      real(kind=c_real), intent(inout) :: qrheti, nrheti
```

Comment:
OK, I see a real problem here and I might have missed it in your previous PRs, so please go back and check.

The fortran reference code has qrheti, nrheti as `intent(out)` but here you have them as `inout`.

Reply:
I noticed that too. I can change the intent in the others as well. In my experience, there were issues with older Fortran compilers and intent(out) variables, but maybe that's all fine now.

Thanks for the eagle eyes!